File: src/main/java/io/github/gravetii/scene/theme/ChangeThemeScene.java
package io.github.gravetii.scene.theme;

import io.github.gravetii.scene.FxDimensions;
import io.github.gravetii.scene.FxScene;
import javafx.geometry.Dimension2D;
import javafx.stage.Stage;

import java.util.Optional;

public class ChangeThemeScene extends FxScene {
  private final ChangeThemeFooterComponent footerComponent;
  private final ChangeThemeComponent themeComponent;

  public ChangeThemeScene(Stage stage) throws Exception {
    super(stage);
    this.footerComponent = new ChangeThemeFooterComponent();
    this.themeComponent = new ChangeThemeComponent(stage, this.footerComponent);
  }

  @Override
  protected void build() {
    this.showCenter(this.themeComponent).showBottom(this.footerComponent);
  }

  @Override
  public String title() {
    return "Change theme";
  }

  @Override
  protected Optional<FxDimensions> preferredDimensions() {
    Dimension2D def = new Dimension2D(800, 550);
    Dimension2D min = new Dimension2D(800, 550);
    Dimension2D max = new Dimension2D(800, 550);
    return Optional.of(new FxDimensions(def, min, max));
  }
}

File: src/main/kotlin/io/github/gravetii/scene/theme/ChangeThemeScene.kt
package io.github.gravetii.scene.theme

import io.github.gravetii.scene.FxDimensions
import io.github.gravetii.scene.FxScene
import javafx.geometry.Dimension2D
import javafx.stage.Stage

class ChangeThemeScene(stage: Stage) : FxScene(stage) {

    private val component = ChangeThemeComponent(stage)

    override fun build() {
        showCenter(component)
    }

    override fun title() = "Change theme"

    override fun preferredDimensions(): FxDimensions {
        val def = Dimension2D(800.0, 550.0)
        return FxDimensions(def, def, def)
    }

}